src/main/java/mil/dds/anet/database/PersonDao.java

```diff
@@ -209,5 +226,25 @@ public Void inTransaction(Handle conn, TransactionStatus status) throws Exceptio
 		});
 		return true;	
 	}
-	
+
+	public CompletableFuture<List<PersonPositionHistory>> getPositionHistory(Map<String, Object> context, Person person) {
+		return new ForeignKeyFetcher<PersonPositionHistory>()
+				.load(context, "person.personPositionHistory", person.getId())
+				.thenApply(l ->
+		{
+			// Derive start and end times; assumes list is in chronological order
+			PersonPositionHistory pphPrev = null;
+			for (final PersonPositionHistory pph : l) {
+				pph.setStartTime(pph.getCreatedAt());
+				if (pphPrev != null) {
+					pphPrev.setEndTime(pph.getStartTime());
+				}
+				pphPrev = pph;
+			}
+			// Remove all null entries
+			l = l.stream().filter(pph -> (pph != null && pph.getPerson() != null)).collect(Collectors.toList());
+			return l;
+		});
```

Comment:
This inner block appears quite similar to the inner block in `PositionDao.getPositionHistory`, extract into a common method?

Reply:
Yes, I saw that, but I had no idea where such a common method should be placed. Should I add it to the PersonPositionHistory bean maybe? Or make a specific util for it?

Reply:
`PersonPositionHistory.java` seems like a good place.